pkg/image/digest.go

```diff
@@ -0,0 +1,13 @@
+package image
+
+import "github.com/opencontainers/go-digest"
+
+// getPartialDigest returns the first 6 characters of a digest
+// if valid.
+func getPartialDigest(d string) (string, error) {
+	_, err := digest.Parse(d)
+	if err != nil {
+		return "", err
+	}
+	return d[7:13], nil
```

Comment:
Given the input is just a string, probably should add a bit more context here to clarify things. This is an image digest with the format `sha256:xxxxxx`. So from 0:6 is `sha256:`, digest prefix and the rest of the SHA digest which you attend to get the first 6 characters. Plus should we check the length first here to ensure it has more than 13 characters to avoid out of bound error even though normally the SHA256 digest should be sufficiently longer than 6 characters?

Reply:
Added more context with an example of the intended result. Also added the length check, while I agree the digest validation should catch this, better safe than to panic.